frodo.js

```diff
@@ -1,55 +1,52 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.